csharp/src/IceSSL/Endpoint.cs

```diff
@@ -6,219 +6,159 @@
+        public override int Timeout => _delegate.Timeout;
+        public override Ice.EndpointType Type => _delegate.Type;
+        public override Ice.Endpoint Underlying => _delegate;
+
+        private readonly Ice.Endpoint _delegate;
+        private readonly Instance _instance;
 
-        public override IceInternal.Endpoint Timeout(int timeout)
+        public override bool Equals(Ice.Endpoint? other)
         {
-            if (timeout == _delegate.Timeout())
+            if (ReferenceEquals(this, other))
             {
-                return this;
+                return true;
             }
-            else
-            {
-                return new Endpoint(_instance, _delegate.Timeout(timeout));
-            }
-        }
-
-        public override string ConnectionId() => _delegate.ConnectionId();
-
-        public override IceInternal.Endpoint ConnectionId(string connectionId)
-        {
-            if (connectionId.Equals(_delegate.ConnectionId()))
+            if (other is Endpoint sslEndpoint)
             {
-                return this;
+                return _delegate.Equals(sslEndpoint._delegate);
             }
             else
             {
-                return new Endpoint(_instance, _delegate.ConnectionId(connectionId));
+                return false;
             }
         }
 
-        public override bool Compress() => _delegate.Compress();
+        public override int GetHashCode() => _delegate.GetHashCode();
+
+        public override string OptionsToString() => _delegate.OptionsToString();
 
-        public override IceInternal.Endpoint Compress(bool compress)
+        public override bool Equivalent(Ice.Endpoint endpoint)
         {
-            if (compress == _delegate.Compress())
+            if (endpoint is Endpoint sslEndpoint)
             {
-                return this;
+                return _delegate.Equivalent(sslEndpoint._delegate);
             }
             else
             {
-                return new Endpoint(_instance, _delegate.Compress(compress));
+                return false;
             }
         }
 
-        public override bool Datagram() => _delegate.Datagram();
-        public override bool Secure() => _delegate.Secure();
-        public override IceInternal.ITransceiver? Transceiver() => null;
+        public override void IceWriteImpl(Ice.OutputStream ostr) => _delegate.IceWriteImpl(ostr);
 
-        private sealed class EndpointI_connectorsI : IceInternal.IEndpointConnectors
-        {
-            public EndpointI_connectorsI(Instance instance, string host, IceInternal.IEndpointConnectors cb)
-            {
-                _instance = instance;
-                _host = host;
-                _callback = cb;
-            }
+        public override Ice.Endpoint NewCompressionFlag(bool compressionFlag)
+            => compressionFlag == _delegate.HasCompressionFlag ? this :
+                new Endpoint(_instance, _delegate.NewCompressionFlag(compressionFlag));
```

Comment:
shouldn't `=>` be in line above here?

```suggestion
        public override Ice.Endpoint NewCompressionFlag(bool compressionFlag) =>
            compressionFlag == _delegate.HasCompressionFlag ? this :
                new Endpoint(_instance, _delegate.NewCompressionFlag(compressionFlag));
```

Reply:
Fixed.